File: CrackingTheCodingInterview/TechnicalQuestions/PascalsTriangle.js
var generate = function(numRows) {
    const triangle = [];
    const firstRow = [1];
    triangle.push(firstRow);
    
    let row = 1;
    while (row < numRows) {
        let currRow = [];
        let prevRow = triangle[row-1];
        
        currRow.push(1)
        
        for (let col = 1; col < prevRow.length; col++) {
            currRow.push(prevRow[col-1] + prevRow[col]);
        }
        
        currRow.push(1);
        
        triangle.push(currRow);
        row++;
    }
    
    return triangle
};

File: CrackingTheCodingInterview/TechnicalQuestions/PascalsTriangle.java
package CrackingTheCodingInterview.TechnicalQuestions;

public class PascalsTriangle {
    public static void main(String[] args) {
        
    }

    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> firstRow = new ArrayList<>();
        
        firstRow.add(1);
        res.add(firstRow);
        
        if (numRows == 1) return res;
        
        
        int row = 1;
        while (row < numRows) {
            List<Integer> currRow = new ArrayList<>();
            List<Integer> prev = res.get(row-1);
            currRow.add(1);
            
            for (int col = 1; col < prev.size(); col++) {
                currRow.add(prev.get(col-1) + prev.get(col));
            }
            
            currRow.add(1);
            res.add(currRow);
            row++;
        }
        
        return res;
    }
}